hack/demo/demo.sh

```diff
@@ -0,0 +1,242 @@
+  httpd:2 -Bbn "${REGISTRY_USERNAME}" "${REGISTRY_PASSWORD}" >"${REGISTRY_AUTH_DIR}/htpasswd"
+
+docker container rm -fv registry || true
+docker container run --rm -d \
+  --name registry \
+  --network kind \
+  -v "${REGISTRY_CERTS_DIR}":/certs \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/ca.crt \
+  -e REGISTRY_HTTP_TLS_KEY=/certs/ca.key \
+  -v "${REGISTRY_AUTH_DIR}":/auth \
+  -e "REGISTRY_AUTH=htpasswd" \
+  -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+  -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+  -p "${REGISTRY_PORT}":"${REGISTRY_PORT}" \
+  registry:2
+
+docker image pull nginx:latest
+docker image tag docker.io/library/nginx:latest "127.0.0.1:${REGISTRY_PORT}/library/nginx:latest"
+
+echo "${REGISTRY_PASSWORD}" | docker login -u "${REGISTRY_USERNAME}" --password-stdin "127.0.0.1:${REGISTRY_PORT}"
+docker image push "127.0.0.1:${REGISTRY_PORT}/library/nginx:latest"
+
+# Retag and push with a tag that doesn't exist in docker.io to test Containerd mirror config
+docker image tag docker.io/library/nginx:latest "127.0.0.1:${REGISTRY_PORT}/library/nginx:$(whoami)"
+docker image push "127.0.0.1:${REGISTRY_PORT}/library/nginx:$(whoami)"
+
+docker build -f "hack/demo/Dockerfile.demo-node" -t mesosphere/kind-node-demo-credential-provider:v1.25.4 .
+
+export KUBECONFIG="${DEMODATA_DIR}"/bootstrap-kubeconfig
+
+kind delete clusters image-credential-provider-bootstrap || true
+kind create cluster --name image-credential-provider-bootstrap --image mesosphere/kind-node:v1.25.4 --config <(
+  cat <<EOF
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+containerdConfigPatches:
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
+    endpoint = ["https://mirror.gcr.io"]
+- |-
+  [plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]
+    username = "${DOCKER_HUB_USERNAME:-}"
+    password = "${DOCKER_HUB_PASSWORD:-}"
+nodes:
+- role: control-plane
+  extraMounts:
+    - hostPath: /var/run/docker.sock
+      containerPath: /var/run/docker.sock
+EOF
+)
+
+# Enable the experimental Cluster topology and cluster resource set features.
+export CLUSTER_TOPOLOGY=true EXP_CLUSTER_RESOURCE_SET=true
+
+# Initialize the management cluster
+clusterctl init --infrastructure docker --wait-providers
+
+kubectl apply -f "hack/demo/docker-clusterclass.yaml"
+
+kubectl create configmap tigera-operator \
+  --from-file=tigera-operator.json=<(curl -fsSL "https://docs.projectcalico.org/archive/v3.24/manifests/tigera-operator.yaml" |
+    gojq --yaml-input \
+      --slurp \
+      --indent=0 \
+      '[ .[] | select( . != null ) | (select(.kind == "Namespace").metadata.labels={"pod-security.kubernetes.io/enforce": "privileged"}) ]')
+kubectl apply -f <(
+  cat <<EOF
+apiVersion: addons.cluster.x-k8s.io/v1beta1
+kind: ClusterResourceSet
+metadata:
+  name: tigera-operator
+spec:
+  clusterSelector:
+    matchLabels:
+      cni: calico
+  resources:
+    - kind: ConfigMap
+      name: tigera-operator
```

Comment:
To avoid the race you can apply it with a single CRS, the ordering is guaranteed.
```suggestion
    - kind: ConfigMap
      name: tigera-operator
    - kind: ConfigMap
      name: calico-installation
```

Reply:
AHA nice! Thanks!